### PR TITLE
Changes some legacy movespeed systems to modifier movespeed

### PR DESCRIPTION
**Pull request by actioninja**
## About The Pull Request

Changes temperature, damage, fat, and (legacy) hunger slowdown to modifier movespeed slowdown.
This should not have a visible effect in game.

If you remember this existing a while ago and breaking things, this is a fixed version. Turns out modifier movespeed supported exactly what I needed the whole time it was just undocumented.

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -1030,6 +1030,18 @@
 /mob/living/carbon/human/updatehealth()
 	. = ..()
 	dna?.species.spec_updatehealth(src)
+	if(HAS_TRAIT(src, TRAIT_IGNOREDAMAGESLOWDOWN))
+		remove_movespeed_modifier(MOVESPEED_ID_DAMAGE_SLOWDOWN)
+		return
+	var/health_deficiency = max((maxHealth - health), staminaloss)
+	if(health_deficiency >= 40)
+		if(movement_type & FLYING)
```
Comment (nemvar) [inline code/modules/mob/living/carbon/human/human.dm]: This seems really hacky. Shouldn't this be two different movespeed modifiers? One of them only applies to flying and the other doesn't apply to it.

**Comment by Shadowflame909** [comment]
I wonder if this does anything for the movement v
Speed change breaking morphs  since they no longer readjust.

File: code/modules/mob/living/carbon/human/human.dm (hunk 1 above)
Comment (nemvar) [reply code/modules/mob/living/carbon/human/human.dm]: Something like
```
add_movespeed_modifier(id, override=TRUE, multiplicative_slowdown=(health_deficiency / 75, blacklisted_movetypes=FLOATING|FLYING)
add_movespeed_modifier(id, override=TRUE, multiplicative_slowdown=(health_deficiency / 25), movetypes=FLYING)
```
It should work

**Comment by optimumtact** [comment]
conflicts

**Review by ghost — APPROVED** [review]

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -1033,6 +1033,16 @@
...
 	dna?.species.spec_updatehealth(src)
+	if(HAS_TRAIT(src, TRAIT_IGNOREDAMAGESLOWDOWN))
+		remove_movespeed_modifier(MOVESPEED_ID_DAMAGE_SLOWDOWN)
+		return
+	var/health_deficiency = max((maxHealth - health), staminaloss)
+	if(health_deficiency >= 40)
+		add_movespeed_modifier(MOVESPEED_ID_DAMAGE_SLOWDOWN, override = TRUE, multiplicative_slowdown = (health_deficiency / 75), blacklisted_movetypes = FLOATING|FLYING)
+		add_movespeed_modifier(MOVESPEED_ID_DAMAGE_SLOWDOWN_FLYING, override = TRUE, multiplicative_slowdown = (health_deficiency / 25), movetypes = FLOATING)
+	else
+		remove_movespeed_modifier(MOVESPEED_ID_DAMAGE_SLOWDOWN)
```
Comment (AnturK) [inline code/modules/mob/living/carbon/human/human.dm]: Shouldn't this be removing the flying slowdown too ?

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/human.dm (hunk 2 above)
Comment (actioninja) [reply code/modules/mob/living/carbon/human/human.dm]: yep

**Comment by actioninja** [comment]
Alright, been testmerged for a bit and does not seem to be causing issues.

File: code/modules/mob/living/carbon/human/human.dm [hunk 3]
```diff
@@ -1033,6 +1033,17 @@
 /mob/living/carbon/human/updatehealth()
 	. = ..()
 	dna?.species.spec_updatehealth(src)
+	if(HAS_TRAIT(src, TRAIT_IGNOREDAMAGESLOWDOWN))
+		remove_movespeed_modifier(MOVESPEED_ID_DAMAGE_SLOWDOWN)
```
Comment (AnturK) [inline code/modules/mob/living/carbon/human/human.dm]: What about in here ? should that trait also stop flyers from slowing down ?

Comment (actioninja) [reply code/modules/mob/living/carbon/human/human.dm]: yeah, probably should have caught that myself

**Comment by AnturK** [comment]
Let's see how it goes this time.

**Pull request merged**